packages/internal/src/debug.js

```diff
@@ -1,41 +1,54 @@
 // @jessie-check
 
+// See https://github.com/Agoric/agoric-sdk/issues/11844
+// See https://github.com/Agoric/agoric-sdk/issues/11845
 let debugInstance = 1;
 
 /**
- * @param {string} name
+ * @param {string} path
  * @param {boolean | 'verbose'} enable
  */
-export const makeTracer = (name, enable = true) => {
+export const makeTracer = (path, enable = true) => {
+  const sub = (step, subEnable = enable) =>
+    makeTracer(`${path}.${step}`, subEnable);
+  const key = `----- ${path}.${debugInstance} `;
   debugInstance += 1;
-  let debugCount = 1;
-  const key = `----- ${name}.${debugInstance} `;
   // the cases below define a named variable to provide better debug info
   switch (enable) {
     case false: {
       const logDisabled = (..._args) => {};
-      return logDisabled;
+      return harden(Object.assign(logDisabled, { sub }));
     }
     case 'verbose': {
       const infoTick = (optLog, ...args) => {
-        if (optLog.log) {
-          console.info(key, (debugCount += 1), ...args);
+        // Sniff tests such as this are inherently unreliable and smell bad.
```

Comment:
```suggestion
        // XXX Sniff tests such as this are inherently unreliable and smell bad.
```

per [kinds of comments](https://github.com/Agoric/agoric-sdk/wiki/Coding-Style#code-comments):
> `XXX` acknowledges some tech debt. It's not broken and doesn't need to be tracked as something to change, but it should not be emulated.

Reply:
Done